README.md

```diff
@@ -2,36 +2,365 @@
+
+To use local [docker daemon process](https://docs.docker.com/config/daemon/#start-the-daemon-manually) for docker build and publish
+
+```console
+dockerd
+```
+
+**or**
+
+[Create a cloud storage bucket](https://cloud.google.com/ai-platform/docs/getting-started-keras#create_a_bucket) for using [Google Cloud build](https://cloud.google.com/cloud-build) for docker image build and publish.
+
+```console
+BUCKET_NAME="your-bucket-name"
+REGION="us-central1"
+gsutil mb -l $REGION gs://$BUCKET_NAME
+```
+
+7. [Optional] Install [nbconvert](https://nbconvert.readthedocs.io/en/latest/)
+
+This is required only if you are using a notebook file as `entry_point` as shown in [usage guide #4](#usage-guide).
+
+```console
+pip install nbconvert
+```
+
+8. Install latest release
+
+```console
+pip install tensorflow-cloud
+```
+
+## Usage guide
+
+As described in the [high level overview](#high-level-overview) section, the `run` API allows you to train your models at scale on GCP. The [`run`](https://github.com/tensorflow/cloud/blob/master/tensorflow_cloud/run.py#L31) API can be used in four different ways. This is defined by where you are running the API (Terminal vs IPython notebook) and what the `entry_point` parameter value is. `entry_point` is an optional Python script or notebook file path to the file that contains the TensorFlow Keras training code. This is the most important parameter in the API.
+
+
+```python
+run(entry_point=None,
+    requirements_txt=None,
+    distribution_strategy='auto',
+    docker_base_image=None,
+    chief_config='auto',
+    worker_config='auto',
+    worker_count=0,
+    entry_point_args=None,
+    stream_logs=False,
+    docker_image_bucket_name=None,
+    **kwargs)
+```
+
+**1. Using a python file as `entry_point`.**
+
+If you have your `tf.keras` model in a python file (`mnist_example.py`), then you can write the following simple script (`scale_mnist.py`) to scale your model on GCP.
+
+```python
+import tensorflow_cloud as tfc
+tfc.run(entry_point='mnist_example.py')
+```
+
+**2. Using a notebook file as `entry_point`.**
+
+If you have your `tf.keras` model in a notebook file (`mnist_example.ipynb`), then you can write the following simple script (`sclae_mnist.py`) to scale your model on GCP.
+
+```python
+import tensorflow_cloud as tfc
+tfc.run(entry_point='mnist_example.ipynb')
+```
+
+**3. Using `run` within a python script that contains the `tf.keras` model.**
+
+You can use the `run` API from within your python file that contains the `tf.keras` model (`mnist_scale.py`). 
+
+```python
+import tensorflow_datasets as tfds
+import tensorflow as tf
+import tensorflow_cloud as tfc
+
+tfc.run(entry_point=None,
+	    distribution_strategy='auto',
+	    requirements_txt='tests/testdata/requirements.txt',
+	    chief_config=tfc.MachineConfig(cpu_cores=8,
+							           memory=30,
+							           accelerator_type=tfc.AcceleratorType.NVIDIA_TESLA_P100,
+							           accelerator_count=2),
+	    worker_count=0)
```

Comment:
These few lines appears to need reformatting. For automatic formatting, you could use `black`. https://black.readthedocs.io/en/stable/